src/main/java/org/dbos/apiary/xa/XADBConnection.java

```diff
@@ -0,0 +1,26 @@
+package org.dbos.apiary.xa;
+
+import java.sql.SQLException;
+import java.sql.ResultSet;
+
+public abstract interface XADBConnection {
```

Comment:
You might consider refactoring this class as I mentioned above. Maybe it's doable to put some common function here as the `default` function implementation.

Reply:
Makes sense

Reply:
Moved most of the repeated stuff to ``BaseXAConnection`` which now gets inherited by ``MySQLXAConnection`` and ``PostgresXAConnection``